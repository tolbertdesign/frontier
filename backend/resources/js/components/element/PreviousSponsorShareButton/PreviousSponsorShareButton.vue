<template>
  <div class="share-button flex flex-col">
    <button
      :class="['share-button icon-above text-white shadow rounded-full px-8 font-normal py-2 cursor-pointer inline-block text-lg hover:text-white bg-secondary', config.className]"
      @click="clicked"
    >
      <FontAwesomeIcon
        :icon="config.icon"
        class="text-5xl fa-w-20"
      />
      {{ config.label }}
    </button>
  </div>
</template>

<script>
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

export default {
  name: 'PreviousSponsorShareButton',
  status: 'prototype',
  release: '1.0.0',
  componentns: {
    FontAwesomeIcon,
  },
  props: {
    participants: {
      type: Array,
      default: () => [],
    },
    ga_page_location: {
      type: String,
      default: 'Student Card Header Section',
    },
    ga_site_location: {
      type: String,
      default: 'Parent Dashboard',
    },
  },
  computed: {
    config () {
      return {
        icon: 'users',
        label: this.lang.view_previous_sponsors_button,
        className: 'bg-secondary view-previous-sponsors',
      }
    },
    lang () {
      return this.$store.state.lang
    },
  },
  methods: {
    gaTrack () {
      this.gaEvent(this.ga_site_location, this.ga_page_location, 'Share Button: view-previous-sponsors')
    },
    clicked (event) {
      const participantUserId = this.getSmallestParticipantId()
      this.$router.push({
        name: 'easy-emailer',
        params: {
          participantUserId: participantUserId,
        },
      })
    },
    getSmallestParticipantId () {
      const participantIds = this.participants.map(obj => obj.id)
      return Math.min(...participantIds)
    },
  },
}
</script>
