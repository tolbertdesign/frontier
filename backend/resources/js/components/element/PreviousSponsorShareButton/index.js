import PreviousSponsorShareButton from './PreviousSponsorShareButton.vue'

export default PreviousSponsorShareButton
