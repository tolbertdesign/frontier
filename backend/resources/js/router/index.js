import Vue from 'vue'
import VueRouter from 'vue-router'
import AppContainer from '@/layouts/AppContainer'
import NProgress from 'nprogress'
import ParentDashboard from '@/views/ParentDashboard'
import TeacherDashboard from '@/views/TeacherDashboard'
import EditParticipant from '@/views/EditParticipant'
import EditProfile from '@/views/EditProfile'
import EasyEmailer from '@/views/EasyEmailer/EasyEmailer'
import FinishLine from '@/views/FinishLine'
import store from '@/store'

Vue.use(VueRouter)

const routes = [
  {
    path: '/v3/home',
    component: AppContainer,
    children: [
      {
        path: 'dashboard',
        name: 'parent-dashboard',
        component: ParentDashboard,
      },
      {
        path: 'teacher-dashboard',
        name: 'teacher-dashboard',
        component: TeacherDashboard,
      },
      {
        path: 'edit-participant/:id',
        name: 'edit-participant',
        component: EditParticipant,
      },
      {
        path: 'edit-profile',
        name: 'edit-profile',
        component: EditProfile,
      },
      {
        path: 'easy-emailer/:participantUserId',
        name: 'easy-emailer',
        component: EasyEmailer,
      },
      {
        path: 'finish-line/:participantUserId',
        name: 'finish-line',
        component: FinishLine,
      },
    ],
  },

]

const router = new VueRouter({
  mode: 'history',
  forcePageRouteRefresh: false,
  linkActiveClass: 'is-active',
  routes,

  scrollBehavior (to, from, savedPosition) {
    return { x: 0, y: 0 }
  },
})

router.beforeEach((routeTo, routeFrom, next) => {
  NProgress.start()
  store.dispatch('setContentGroup', routeTo)

  if (router.options.forcePageRouteRefresh) {
    window.location.href = '/v3/home/dashboard/' + routeTo.path
    router.options.forcePageRouteRefresh = false
  } else {
    next()
  }
})

router.afterEach(() => {
  NProgress.done()
})

export default router
