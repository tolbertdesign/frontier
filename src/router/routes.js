import store from '@/vuex/store'

export default [
  {
    path: '/',
    name: 'welcome',
    component: () => import(/* webpackChunkName: "welcome" */ '@/pages/welcome'),
    meta: {
      layout: 'FullScreen',
      requiresVisitor: true,
    },
  },
  {
    path: '/login',
    name: 'login',
    component: () => import(/* webpackChunkName: "login" */ '@/pages/login'),
    meta: {
      layout: 'FullScreen',
      requiresVisitor: true,
    },
  },
  {
    path: '/register',
    name: 'register',
    component: () => import(/* webpackChunkName: "register" */ '@/pages/register'),
    meta: {
      layout: 'FullScreen',
      requiresVisitor: true,
    },
  },
  {
    path: '/buy',
    name: 'buy',
    component: () => import(/* webpackChunkName: "buy" */ '@/pages/buy'),
    meta: {
      layout: 'FlexboxApp',
    },
  },
  {
    path: '/contact',
    name: 'contact',
    component: () => import(/* webpackChunkName: "contact" */ '@/pages/contact'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/dashboard',
    name: 'dashboard',
    component: () => import(/* webpackChunkName: "dashboard" */ '@/pages/dashboard'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/landing',
    name: 'landing',
    component: () => import(/* webpackChunkName: "landing" */ '@/pages/landing'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/templates',
    name: 'templates',
    component: () => import(/* webpackChunkName: "templates" */ '@/pages/templates'),
    meta: {
      layout: 'Sidebar',
      requiresAuth: true,
    },
  },
  {
    path: '/documents',
    name: 'documents',
    component: () => import(/* webpackChunkName: "documents" */ '@/pages/documents'),
    meta: {
      layout: 'Sidebar',
      requiresAuth: true,
    },
  },
  {
    path: '/reports',
    name: 'reports',
    component: () => import(/* webpackChunkName: "reports" */ '@/pages/reports'),
    meta: {
      layout: 'Sidebar',
      requiresAuth: true,
    },
  },
  {
    path: '/profile',
    name: 'profile',
    component: () => import(/* webpackChunkName: "edit-profile" */ '@/pages/profile'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/settings',
    name: 'settings',
    component: () => import(/* webpackChunkName: "settings" */ '@/pages/settings'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/event/list',
    name: 'event-list',
    component: () => import(/* webpackChunkName: "event-list" */ '@/pages/event-list'),
    props: true,
  },
  {
    path: '/event/create',
    name: 'event-create',
    component: () => import(/* webpackChunkName: "event-create" */ '@/pages/event-create'),
  },
  {
    path: '/event/:id',
    name: 'event-show',
    component: () => import(/* webpackChunkName: "event-show" */ '@/pages/event-show'),
    props: true,
    beforeEnter (routeTo, routeFrom, next) {
      store
        .dispatch('event/fetchEvent', routeTo.params.id)
        .then(event => {
          routeTo.params.event = event
          next()
        })
        .catch(error => {
          if (error.response && error.response.status == 404) {
            next({ name: '404', params: { resource: 'event' } })
          } else {
            next({ name: 'network-issue' })
          }
        })
    },
  },
  {
    path: '/user/:id',
    component: () => import(/* webpackChunkName: "user" */ '@/pages/user'),

  },
  {
    path: '/404',
    name: '404',
    component: () => import(/* webpackChunkName: "not-found" */ '@/pages/not-found'),
  },
  {
    path: '/network-issue',
    name: 'network-issue',
    component: () => import(/* webpackChunkName: "network-issue" */ '@/pages/network-issue'),
  },
  {
    path: '*',
    redirect: {
      name: '404',
      params: {
        resource: 'page',
      },
    },
  },
]
