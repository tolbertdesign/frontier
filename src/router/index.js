import Vue from 'vue'
import Router from 'vue-router'
// Adds a loading bar at the top during page loads.
import NProgress from 'nprogress'
import 'nprogress/nprogress.css'

// https://github.com/declandewet/vue-meta
import Meta from 'vue-meta'
import { sync } from 'vuex-router-sync'
import routes from '@/router/routes'
import store from '@/vuex/store'

Vue.use(Router)
Vue.use(Meta, {
  // The component option name that vue-meta looks for meta info on.
  keyName: 'page',
})

const router = new Router({
  routes,
  base: process.env.BASE_URL,
  // Use the HTML5 history API (i.e. normal-looking routes)
  // instead of routes with hashes (e.g. example.com/#/about).
  // This may require some server configuration in production:
  // https://router.vuejs.org/en/essentials/history-mode.html#example-server-configurations
  mode: 'history',
  // Simulate native-like scroll behavior when navigating to a new
  // route and using back/forward buttons.
  scrollBehavior (to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition
    } else {
      return { x: 0, y: 0 }
    }
  },
})

router.beforeEach((routeTo, routeFrom, next) => {
  // If this isn't an initial page load...
  if (routeFrom.name !== null) {
    // Start the route progress bar.
    NProgress.start()
  }

  if (routeTo.matched.some(record => record.meta.requiresAuth)) {
    if (!isLoggedIn()) {
      next({
        name: 'home',
      })
    } else {
      next()
    }
  } else if (routeTo.matched.some(record => record.meta.requiresVisitor)) {
    if (isLoggedIn()) {
      next({
        name: 'dashboard',
      })
    } else {
      next()
    }
  } else {
    next() // make sure to always call next()!
  }

  // If auth is required and the user is NOT currently logged in,
  // redirect to login.
  // redirectToLogin()

  // function redirectToLogin () {
  //   // Pass the original route to the login component
  //   next({ name: 'login', query: { redirectFrom: routeTo.fullPath } })
  // }
})

router.afterEach(() => {
  NProgress.done()
})

function isLoggedIn () {
  return store.getters['auth/loggedIn']
}

sync(store, router)

export default router
