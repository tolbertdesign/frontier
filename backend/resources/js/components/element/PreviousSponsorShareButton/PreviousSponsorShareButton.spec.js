import { shallowMount, createLocalVue } from '@vue/test-utils'
import PreviousSponsorShareButton from './PreviousSponsorShareButton.vue'
import router from '@/router'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

const localVue = createLocalVue()

const participants = [
  { id: 3 },
  { id: 1 },
  { id: 2 },
]
const ga_page_location = 'ga page location'
const ga_site_location = 'ga site location'

const $store = {
  state: {
    lang: {
      view_previous_sponsors_button: 'view previous sponsors button',
    },
  },
}

describe('PreviousSponsorShareButton', () => {
  it('Snapshop test for PreviousSponsorShareButton', () => {
    const wrapper = shallowMount(PreviousSponsorShareButton, {
      localVue,
      router,
      mocks: {
        $store,
      },
      components: {
        FontAwesomeIcon,
      },
      propsData: {
        participants: participants,
        ga_page_location: ga_page_location,
        ga_site_location: ga_site_location,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('Test getSmallestParticipantId method', () => {
    const wrapper = shallowMount(PreviousSponsorShareButton, {
      localVue,
      router,
      mocks: {
        $store,
      },
      components: {
        FontAwesomeIcon,
      },
      propsData: {
        participants: participants,
        ga_page_location: ga_page_location,
        ga_site_location: ga_site_location,
      },
    })
    expect(wrapper.vm.getSmallestParticipantId()).toBe(1)
  })
})
