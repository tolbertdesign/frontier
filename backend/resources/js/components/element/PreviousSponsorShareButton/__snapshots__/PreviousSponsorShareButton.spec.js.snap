// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousSponsorShareButton Snapshop test for PreviousSponsorShareButton 1`] = `
<div class="share-button flex flex-col"><button class="icon-above text-white shadow rounded-full px-8 font-normal py-2 cursor-pointer inline-block text-lg hover:text-white bg-secondary bg-secondary view-previous-sponsors">
    <fontawesomeicon-stub icon="users" class="text-5xl fa-w-20"></fontawesomeicon-stub>
    view previous sponsors button
  </button></div>
`;
